src/Microsoft.OData.Core/ODataCollectionWriterCore.cs

```diff
@@ -199,10 +200,11 @@ public sealed override void WriteItem(object item)
         /// </summary>
         /// <param name="item">The collection item to write.</param>
         /// <returns>A task instance that represents the asynchronous write operation.</returns>
-        public sealed override Task WriteItemAsync(object item)
+        public sealed override async Task WriteItemAsync(object item)
         {
             this.VerifyCanWriteItem(false);
-            return TaskUtils.GetTaskForSynchronousOperation(() => this.WriteItemImplementation(item));
+            await this.WriteItemImplementationAsync(item)

```

Comment:
Elidable?

Reply:
Explained why it's not particularly recommendable to do that here https://github.com/OData/odata.net/pull/2059#discussion_r635801748